synapse/config/server_notices_config.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 New Vector Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from ._base import Config
+from synapse.types import UserID
+
+DEFAULT_CONFIG = """\
+# Server Notices room configuration
+#
+# Uncomment this section to enable a room which can be used to send notices
+# from the server to users. It is a special room which cannot be left; notices
+# come from a special "notices" user id.
+#
+# If you uncomment this section, you *must* define the system_mxid_localpart
+# setting, which defines the id of the user which will be used to send the
+# notices.
+#
+# It's also possible to override the room name, or the display name of the
+# "notices" user.
+#
+# server_notices:
+#   system_mxid_localpart: notices
+#   system_mxid_display_name: "Server Notices"
+#   room_name: "Server Notices"
+"""
+
+
+class ServerNoticesConfig(Config):
+    def __init__(self):
+        super(ServerNoticesConfig, self).__init__()
+
+        """The MXID to use for server notices.
+
+        None if server notices are not enabled.
+
+        type: str|None
+        """
+        self.server_notices_mxid = None
+
+        """The display name to use for the server notices user.
+
+        None if server notices are not enabled.
+
+        type: str|None
+        """
+        self.server_notices_mxid_display_name = None
+
+        """The name to use for the server notices room.
+
+        None if server notices are not enabled.
+
+        (TODO: i18n)
+
+        type: str|None
+        """
```

Comment:
Aren't these usually documented under `Attributes` in the class level docstring? I'm finding it hard to read when done inline if I'm honest

Reply:
Yeah, it's not the easiest, but it made my IDE pick up the type right. sigh.